src/test/java/bio/terra/service/snapshotbuilder/query/QueryTest.java

```diff
@@ -25,4 +28,89 @@ public static Query createQuery() {
   void renderSQL() {
     assertThat(createQuery().renderSQL(null), is("SELECT t.* FROM table AS t"));
   }
+
+  @Test
+  void renderComplexSQL() {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+
+    TablePointer conditionOccurrencePointer = TablePointer.fromTableName("condition_occurrence");
+    TableVariable conditionOccurrenceVariable =
+        TableVariable.forJoined(
+            conditionOccurrencePointer,
+            "person_id",
+            new FieldVariable(new FieldPointer(tablePointer, "person_id"), tableVariable));
+
+    TablePointer conditionAncestorPointer = TablePointer.fromTableName("condition_ancestor");
+    TableVariable conditionAncestorVariable =
+        TableVariable.forJoined(
+            conditionAncestorPointer,
+            "ancestor_concept_id",
+            new FieldVariable(
+                new FieldPointer(conditionOccurrencePointer, "condition_concept_id"),
+                conditionOccurrenceVariable));
+
+    Query query =
+        new Query(
+            List.of(
+                new FieldVariable(
+                    new FieldPointer(tablePointer, "person_id", "COUNT"),
+                    tableVariable,
+                    null,
+                    true)),
+            List.of(tableVariable, conditionOccurrenceVariable, conditionAncestorVariable),
+            new BooleanAndOrFilterVariable(
+                BooleanAndOrFilterVariable.LogicalOperator.AND,
+                List.of(
+                    new BooleanAndOrFilterVariable(
+                        BooleanAndOrFilterVariable.LogicalOperator.OR,
+                        List.of(
+                            new BinaryFilterVariable(
+                                new FieldVariable(
+                                    new FieldPointer(
+                                        conditionOccurrencePointer, "condition_concept_id"),
+                                    conditionOccurrenceVariable),
+                                BinaryFilterVariable.BinaryOperator.EQUALS,
+                                new Literal(316139)),
```

Comment:
This is fine although you could create a test-local helper method to remove some of the duplication, if you wanted to.

Not sure how much work we should do on this as part of the API yet since I'm not sure exactly how we'll be using it.